### PR TITLE
Modify list level

### PR DESCRIPTION
**Pull request by ruiliuinin**

### CODE REVIEW

File: build/squire-raw.js [hunk 1]
```diff
@@ -180,7 +180,7 @@ TreeWalker.prototype.previousPONode = function () {
     }
 };
 
-var inlineNodeNames  = /^(?:#text|A(?:BBR|CRONYM)?|B(?:R|D[IO])?|C(?:ITE|ODE)|D(?:ATA|EL|FN)|EM|FONT|HR|I(?:FRAME|MG|NPUT|NS)?|KBD|LI|[OU]L|Q|R(?:P|T|UBY)|S(?:AMP|MALL|PAN|TR(?:IKE|ONG)|U[BP])?|U|VAR|WBR)$/;
```
Comment (ruiliuinin) [inline build/squire-raw.js:183]: Revert the changes of adding LI/OL/UL as inlineNodeNames which caused issues in bulleted lists after pressing Enter

File: build/squire-raw.js [hunk 2]
```diff
@@ -1099,6 +1099,17 @@ var moveRangeBoundariesDownTree = function ( range ) {
         range.setStart( startContainer, startOffset );
         range.setEnd( endContainer, endOffset );
     }
+
```
Comment (ruiliuinin) [inline build/squire-raw.js:1102]: Fix the cursor missing issue in Firefox after the img element in text is deleted. 
Found this issue when deleting the tab span in Firefox and noticed the img element has the same problem. The tab span will only be used in CEF, so it is not a problem for tab anymore. Just leave this fix here for img since I already got a fix for it.

File: build/squire-raw.js [hunk 3]
```diff
@@ -1322,7 +1333,12 @@ var onKey = function ( event ) {
         this._updatePath( range, true );
     }
 
-    this.fireEvent( 'postKeyDown', event );
+    var postKeyDownEvent = {
+        ctrlKey: event.ctrlKey,
+        altKey: event.altKey,
+        which: event.which
+    };
+    this.fireEvent( 'postKeyDown', postKeyDownEvent );
```
Comment (ruiliuinin) [inline build/squire-raw.js:1341]: Fix the error  "Cannot assign to read only property 'type' of object '[object KeyboardEvent]'" when use the keyDown event directly for the poseKeyDown event. I got this error on my local. Although it seems the build on the server doesn't have the problem. It seems to be save to not directly use the keydown event.

**Comment by ruiliuinin** [comment]
Added the function insertTabIndent for the tab issue. Also include the fixes for some other issues I noticed. Please let me know if we don't want the fixes for other issues in this change list.

File: build/squire-raw.js [hunk 4]
```diff
@@ -1099,6 +1099,17 @@ var moveRangeBoundariesDownTree = function ( range ) {
...
+
+    // When the endOffset is at beginning of a node while the startOffset is at the end of 
+    // the previous sibling node, set the end of the range to be the same as the start. 
+    // This could solve the cursor missing problem in Firefox when the IMG element surranded 
+    // by text is deleted. 
+    if ( range.startContainer === range.commonAncestorContainer.childNodes[0] &&
+         range.startOffset === range.commonAncestorContainer.childNodes[0].length && 
+         range.endContainer === range.commonAncestorContainer.childNodes[1] && 
+         range.endOffset === 0 ) {
+        range.setEnd( range.startContainer, range.startOffset );
```
Comment (akauffmanGG) [inline build/squire-raw.js]: I think you can just use range.collapse(true) here.

File: build/squire-raw.js [hunk 5]
```diff
@@ -4246,6 +4262,18 @@ proto.insertElement = function ( el, range ) {
     return this;
 };
 
+proto.insertTabIndent = function( ) {
```
Comment (akauffmanGG) [inline build/squire-raw.js]: spacing

File: build/squire-raw.js [hunk 6]
```diff
@@ -4246,6 +4262,18 @@ proto.insertElement = function ( el, range ) {
...
+    var tabElement, textElement;
+    tabElement = this.createElement( 'SPAN', {
+        style: 'letter-spacing: 40px;'
+    } );
```
Comment (akauffmanGG) [inline build/squire-raw.js]: spacing
